### PR TITLE
Explain behavior of _

### LINKED ISSUE #31154: Document how the "_" pattern works
#31153 was closed and the user who filed it was educated, but they're not the first and not the last person to be confused by this. AFAIK the behavior of `_` &mdash; especially how it's different from `_foo` &mdash; is not documented, at least I couldn't find anything in the reference and it seems a tad obscure to be mentioned in the book.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #31154

### CODE REVIEW

**Comment by durka** [comment]
What about temporaries being dropped?

**Comment by steveklabnik** [comment]
That's a consequence of this behavior, I guess it does deserve its own explanation too.

File: src/doc/book/patterns.md [hunk 1]
```diff
@@ -173,7 +173,31 @@ let (x, _, z) = coordinate();
...
+let tuple = (5, String::from("five"));
+
+// Here, tuple is _not_ moved, as the String was never moved, and u32 is Copy:
+let (x, _) = tuple;
+
+// That means this works:
+println!("Tuple is: {:?}", tuple);
+```
+
+In a similar fashion to `_`, you can use `..` in a pattern to disregard
```
Comment (petrochenkov) [inline src/doc/book/patterns.md]: s/in a pattern/in some patterns
`..` unlike `_` is not an independent pattern, it's part of some other pattern's syntax

**Comment by steveklabnik** [comment]
Updated to address both comments.

**Comment by durka** [comment]
:+1:

**Pull request merged**